mne/viz/evoked.py

```diff
@@ -1191,3 +1191,342 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+
+    # let's plot!
+    if ax is None:
+        fig, ax = plt.subplots(1, 1)
+        fig.set_size_inches(8, 6)
+
+    # style the individual condition time series
+    if not isinstance(colors, dict):
+        colors_ = ['#e41a1c', '#377eb8', '#4daf4a', '#984ea3', '#ff7f00',
+                   '#1b9e77', '#d95f02', '#7570b3', '#e7298a', '#66a61e']
+        colors = dict((condition, color) for condition, color
+                      in zip(conditions, colors_))
+    else:
+        style_error = "Condition {} could not be mapped to a color."
+        for condition in conditions:
+            if condition not in colors:
+                if "/" not in condition:
+                    raise ValueError(style_error.format(condition))
+                for color in colors:
+                    if color in condition.split("/"):
+                        colors[condition] = colors[color]
+                        break
+
+    if not isinstance(linestyles, dict):
+        linestyles = dict((condition, linestyle) for condition, linestyle in
+                          zip(conditions, ['-'] * len(conditions)))
+    else:
+        style_error = "Condition {} could not be mapped to a linestyle."
+        for condition in conditions:
+            if condition not in linestyles:
+                if "/" not in condition:
+                    raise ValueError(style_error.format(condition))
+                for linestyle in linestyles:
+                    if linestyle in condition.split("/"):
+                        linestyles[condition] = linestyles[linestyle]
+                        break
+
+    if styles is None:
+        styles = dict()
+    for condition, color, linestyle in zip(conditions, colors, linestyles):
+        styles[condition] = styles.get(condition, dict())
+        styles[condition]['c'] = styles[condition].get('c', colors[condition])
+        styles[condition]['linestyle'] = styles[condition].get(
+            'linestyle', linestyles[condition])
+
+    # the actual plot
+    any_negative, any_positive = False, False
+    for condition in conditions:
+        if picks is not None:
+            if ch_type == 'grad':
+                d = ((_merge_grad_data(evokeds[condition]
+                      .data[pairpicks, :][picks, :])).T * scaling).mean(-1)
+            else:
+                d = ((evokeds[condition].data[picks, :]).T * scaling).mean(-1)
+        else:
+            d = ((evokeds[condition].data).T * scaling).std(-1)
+        if ci and picks is not None:
+            sem_ = sem_array[condition]
+            ax.fill_between(times, sem_[0].flatten() * scaling,
+                            sem_[1].flatten() * scaling,
+                            color=styles[condition]['c'], alpha=.333)
+
+        ax.plot(times, d, zorder=10, label=condition, **styles[condition])
+        if any(d > 0):
+            any_positive = True
+        if any(d < 0):
+            any_negative = True
+
+    # truncate the y axis ... yes, it's that complicated
```

Comment:
maybe a separate private function for this then? might also be used elsewhere then